apps/sched/sched.js

```diff
@@ -84,6 +84,9 @@ function showAlarm(alarm) {
     let date = new Date(alarm.date);
     let rp = alarm.rp;
     switch(rp.interval) {
+      case true:
```

Comment:
Note that the switch is on "rp.interval", which is a new field (e.g. rp : { interval: "month", num: 2 } ).
So this line 87 will not account for rp: true (as far as I know, there was no behavior defined for rp: true for an alarm with a date), but it will account for rp : { interval: true }.
({rp: true}).interval == undefined, so the previous code would just have left the date as before for rp : true.

Reply:
I've attempted to track this in some typings (#2640), am I right in thinking if we've got a date, we'll have `rp: { interval, num }`, and otherwise we've not got `.date`, so we'll have `rp: true / false` ?
Or could there be cases where we overlap still? (eg `{ date: "...", rp: true }`)

Unfortunately I've not been able to shoehorn `tsc` into typechecking JS successfully yet, so if we do get contributions later on we'll not be able to use the types to check these things just yet.

Reply:
You're right, the Alarm UI supports the case without overlap, but the latest commit also supports that, if a given custom app puts { date: "...", rp: true }, the alarm will just be repeated everyday from that day on.
For the other cases, the [readme ](https://github.com/espruino/BangleApps/tree/master/apps/sched) of `sched` is up to date.